packages/server/src/client/internal/open-feature-client.ts

```diff
@@ -279,35 +281,38 @@ export class OpenFeatureClient implements Client {
       logger: this._logger,
     };
 
-    try {
-      const frozenContext = await this.beforeHooks(allHooks, hookContext, options);
+    const evaluationDetails = await(async () => {
```

Comment:
I'm totally OK with the logic here, but I have a very slight concern that this could be creating a large amount of memory churn because we will be creating this anon function (and needing to clean it up) at every flag evaluation.

Reply:
If you added this IIFE just to avoid a mutable var, I would just add the mutable var.

Reply:
I also think it might be more simple to just accept the mutation here. Additionally as @toddbaert say we can try to save some memory allocs here, even though it might be okay.

Reply:
@lukas-reining, could you please quickly re-review to see if you prefer this change? Thanks!